js/live-stream-embed.js

```diff
@@ -0,0 +1,77 @@
+// Live Stream Video
+
+var ipfs_gateway_self = 'http://ipfs-mirror-0.mesh.world:8080'; // IPFS gateway of this node
+var ipfs_gateway_origin = 'http://ipfs-mirror-0.mesh.world:8080'; // IPFS gateway of origin stream
+var m3u8_ipfs = 'http://ipfs-mirror-0.mesh.world/live.m3u8'; // File path to m3u8 with IPFS content via HTTP server
+// var m3u8_ipfs='__IPFS_GATEWAY_ORIGIN__/ipns/__IPFS_ID_ORIGIN__'; // URL to m3u8 via IPNS (uncomment to enable)
+var m3u8_http_urls = ['http://ipfs-mirror-0.mesh.world/live.m3u8']; // Optional list of URLs to m3u8 over HTTP
+
+var live = videojs('live');
+
+// For any browser except Safari
+if (/^((?!chrome|android).)*safari/i.test(navigator.userAgent) === false) {
+  // Override native player for platform and browser consistency
+  videojs.options.html5.nativeAudioTracks = false;
+  videojs.options.html5.nativeVideoTracks = false;
+  videojs.options.hls.overrideNative = true;
+}
+
+function httpStream() {
+  live.src({
+    src: m3u8_http_urls[Math.floor(Math.random() * m3u8_http_urls.length)],
+    type: 'application/x-mpegURL'
+  });
+  loadStream();
+}
+
+function ipfsStream() {
+  live.src({
+    src: m3u8_ipfs,
+    type: 'application/x-mpegURL'
+  });
+  loadStream();
+  videojs.Hls.xhr.beforeRequest = function(options) {
+    // Replace IPFS gateway of origin with that of this node
+    options.uri = options.uri.replace(ipfs_gateway_origin, ipfs_gateway_self);
+    if (options.uri.indexOf('/ipfs/')) {
+      document.getElementById('loadingTitle').innerHTML = 'Located stream via IPFS';
+      document.getElementById('msg').innerHTML = 'Downloading video content...';
+    }
+    console.debug(options.uri);
+    return options;
+  };
+}
+
+function loadStream() {
+  document.getElementById('LoadingStream').style.display = 'block';
+  document.getElementById('SelectStream').style.display = 'none';
+}
+
+document.querySelector('.ipfs-stream').addEventListener('click', function(event) {
+  ipfsStream();
+});
+
+document.querySelector('.http-stream').addEventListener('click', function(event) {
+  httpStream();
+});
+
+live.metadata = 'none';
+
+live.on('loadedmetadata', function() {
+  document.getElementById('StreamSelector').style.display = 'none';
+});
+
+live.on('loadeddata', function(event) {
+  console.debug(event);
+});
+
+live.on('error', function(event) {
+  console.debug(this.error());
+  document.getElementById('loadingTitle').innerHTML = 'Unable to load live stream';
```

Comment:
Do you want to add a way to refresh? (rn hitting back doesn't work, just a hard page reload I think?)

Reply:
Added a refresh button on error in https://github.com/ournetworks/ournetworks.ca/pull/58/commits/9b74b4a1c57d2076536cc0ef3ec1bfaf2b2575e1